src/services/shopify-client.test.ts

```diff
@@ -0,0 +1,223 @@
+    expect(result.success).toBe(true);
+    expect(result.value[0].shopName).toEqual(connectedWithListSyncResponse.data[0].shopName);
+    expect(result.value[0].list.name).not.toBe(undefined);
+    expect(result.value[0].list.amountSubscribers).not.toBe(undefined);
+    expect(result.value[0].list.state).toEqual(SubscriberListState.synchronizingContacts);
+  });
+
+  it('should set connected shop with list connected but synchronizing contacts even if sync flag is bool', async () => {
+    // Arrange
+    const connectedWithListSyncResponseBool = {
+      data: [
+        {
+          connectedOn: '2019-08-09T15:13:52.262Z',
+          dopplerListId: '27311899',
+          dopplerListName: 'Shopify Contacto',
+          importedCustomersCount: '6',
+          shopName: 'dopplerdevshop.myshopify.com',
+          shopifyAccessToken: '2741255a41f37341d0fa6d64d58e5c86',
+          syncProcessDopplerImportSubscribersTaskId: 'import-100544891',
+          syncProcessInProgress: true,
+          syncProcessLastRunDate: '2019-08-09T15:24:56.183Z',
+        },
+      ],
+    };
+    const request = jest.fn(async () => connectedWithListSyncResponseBool);
+    const shopifyClient = createHttpShopifyClient({ request });
+
+    // Act
+    const result = await shopifyClient.getShopifyData();
+
+    // Assert
+    expect(request).toBeCalledTimes(1);
+    expect(result).not.toBe(undefined);
+    expect(result.success).toBe(true);
+    expect(result.value[0].shopName).toEqual(connectedWithListSyncResponseBool.data[0].shopName);
+    expect(result.value[0].list.name).not.toBe(undefined);
+    expect(result.value[0].list.amountSubscribers).not.toBe(undefined);
+    expect(result.value[0].list.state).toEqual(SubscriberListState.synchronizingContacts);
+  });
+
+  it('should set not connected shop', async () => {
+    // Arrange
+    const notConnected = {
+      data: [],
+    };
+    const request = jest.fn(async () => notConnected);
+    const shopifyClient = createHttpShopifyClient({ request });
+
+    // Act
+    const result = await shopifyClient.getShopifyData();
+
+    // Assert
+    expect(request).toBeCalledTimes(1);
+    expect(result).not.toBe(undefined);
+    expect(result.success).toBe(true);
+    expect(result.value).toEqual([]);
+  });
+
+  it('should set error connecting shopify', async () => {
+    // Arrange
+    const request = jest.fn(async () => {});
+    request.mockImplementation(() => {
+      throw new Error();
+    });
+    const shopifyClient = createHttpShopifyClient({ request });
+    const consoleError = console.error;
+    console.error = jest.fn(); // silence console error for this test run only
+
+    // Act
+    const result = await shopifyClient.getShopifyData();
+
+    // Assert
+    expect(request).toBeCalledTimes(1);
+    expect(result).not.toBe(undefined);
+    expect(result.success).toBe(false);
+    expect(result.error).not.toBe(undefined);
+    console.error = consoleError; // Restore console error logs
```

Comment:
@andresmoschini @fcoronelmakingsense I fixed this issue with `console.error`. Now if I use `console.error` bellow it works as it should.

Reply:
It should be in `aftereach` because if the assertion fails, it is never executed.